samigo/samigo-app/src/webapp/jsf/delivery/item/deliverMatching.jsp

```diff
@@ -38,20 +38,10 @@ should be included in file importing DeliveryMessages
   </t:dataList>
 
   <h:dataTable value="#{question.matchingArray}" var="matching">
-    <h:column rendered="#{delivery.feedback eq 'true' &&
-       delivery.feedbackComponent.showCorrectResponse && !delivery.noFeedback=='true'}">
-      <h:panelGroup id="image"
-        rendered="#{matching.isCorrect}"
-        styleClass="si si-check-lg" >
-      </h:panelGroup>
-      <h:panelGroup id="ximage"
-        rendered="#{!matching.isCorrect}"
-        styleClass="si si-remove feedBackCross" >
-      </h:panelGroup>
-      <h:graphicImage id="image2"
-        width="16" height="16"
-        alt="#{deliveryMessages.alt_incorrect}" url="/images/delivery/spacer.gif">
-      </h:graphicImage>
+    <h:column rendered="#{delivery.feedback eq 'true' && delivery.feedbackComponent.showCorrectResponse && !delivery.noFeedback=='true'}">
+      <h:panelGroup id="image" rendered="#{matching.isCorrect eq 'true'}" styleClass="si si-check-lg" > </h:panelGroup>
+      <h:panelGroup id="ximage" rendered="#{not empty matching.isCorrect && matching.isCorrect eq 'false'}" styleClass="si si-remove feedBackCross"> </h:panelGroup>
```

Comment:
i guess matching.isCorrect eq 'false' is all that is needed right @ern? it is a Boolean behind the scenes so it can be null

Reply:
I think if it's a `Boolean `and not a `boolean `you may have a `not empty matching.isCorrect` on both of these conditions to avoid an NPE? I'm not too sure about JSF.

Reply:
no doesn't seem to NPE

the root of the problem seems to be this: `!matching.isCorrect` which seems to match null and false